clang/lib/CIR/CodeGen/CIRGenCall.cpp

```diff
@@ -738,6 +738,9 @@ RValue CIRGenFunction::buildCall(const CIRGenFunctionInfo &CallInfo,
         buildCallLikeOp(*this, callLoc, indirectFuncTy, indirectFuncVal,
                         directFuncOp, CIRCallArgs, InvokeDest);
 
+    callLikeOp->setAttr("extra_attrs", mlir::cir::ExtraFuncAttributesAttr::get(
+                                           builder.getContext(), Attrs));
```

Comment:
The point about creating the `createCallOp` helpers, is to avoid setting the args in adhoc fashion like this. You should instead pass a `ExtraFuncAttributesAttr` to `buildCallLikeOp`, and change `buildCallLikeOp` to invoke the helpers you introduced.